packages/qdrant-loader/src/qdrant_loader/core/pipeline/source_processor.py

```
@@ -71,8 +72,15 @@ async def process_source_type(
 
                 # Use the connector as an async context manager to ensure proper initialization
                 async with connector:
-                    # Get documents from this source
-                    documents = await connector.get_documents()
+                    documents = []
+                    if isinstance(connector, BaseConnector):
+                        try:
+                            async for document in connector.stream_documents():
+                                documents.append(document)
```

Ruff (0.15.14)

[warning] Use `list.extend` with an async comprehension to create a transformed list

(PERF401)